### PR TITLE
Add rounded rectangular cylinder geometry primitive

### PR DESCRIPTION
**Pull request by mstoeckl**
This pull request introduces a new rounded rectangular cylinder geometry, with interface modelled after `EGS_EllipticCylindersT`. So far, I've used it to model the roundings of milled slots/rectangular holes more accurately, and because a set of concentric rectangles is often simpler and cheaper than voxels or an `XYZGeometry`. 

It appears to pass the `EGS_GeometryTester`, but there may still be edge cases from floating point effects lurking (for instance, one test gets a single almost-not failure at ~10M trials). I also suspect one or two very minor bugs that most likely lie in the interaction between the rounded rectangles and the various composite geometries.

The included test geometry requires the range selection modification of #76 to remain reasonably simple.

Copyright assignment is as before.

### CODE REVIEW

**Comment by mstoeckl** [comment]
There currently appears to be a corner-case `howfar` bug; it will be fixed soon.

**Comment by mstoeckl** [comment]
I found and fixed a minor floating point bug in the `howfar` code, which triggers most often when the x- and y- components of the direction vector are equal, the rounded rectangle is a square, and floating point errors
conspire to push the intersection point with the sides of the square further out than would be possible
at infinite precision. I have yet to find any other issues related to the rounded rectangle geometry. [The `EGS_NDGeometry` bug I have a fix for, but it needs more testing.]

File: HEN_HOUSE/egs++/geometry/egs_roundrect_cylinders/egs_roundrect_cylinders.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#
+###############################################################################
+#
+#  A set of concentric rounded rectangular cylinders.
+#
+###############################################################################
+*/
+
+
+/*! \file egs_cylinders.cpp
```
Comment (rtownson) [inline HEN_HOUSE/egs++/geometry/egs_roundrect_cylinders/egs_roundrect_cylinders.cpp]: File should be egs_roundrect_cylinders.cpp

File: HEN_HOUSE/egs++/geometry/egs_roundrect_cylinders/egs_roundrect_cylinders.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+#  FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for
+#  more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with EGSnrc. If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+#
+#  Author:
```
Comment (rtownson) [inline HEN_HOUSE/egs++/geometry/egs_roundrect_cylinders/egs_roundrect_cylinders.cpp]: Author should be `Manuel Stoeckl`

File: HEN_HOUSE/egs++/geometry/egs_roundrect_cylinders/egs_roundrect_cylinders.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                egsWarning("createGeometry(round rectangular cylinders): missing/wrong "
+                           "'y-axis' input\n");
+                return 0;
+            }
+            g = new EGS_RoundRectCylindersT<EGS_Projector,EGS_Projector>(x_widths,y_widths,radii,xo,"",
+                    EGS_Projector(EGS_Vector(ax[0],ax[1],ax[2]),"Any"),
+                    EGS_Projector(EGS_Vector(ay[0],ay[1],ay[2]),""));
+        }
+        g->setName(input);
+        g->setMedia(input);
```
Comment (rtownson) [inline HEN_HOUSE/egs++/geometry/egs_roundrect_cylinders/egs_roundrect_cylinders.cpp]: Add a line here to set region labels:
`g->setLabels(input);`

**Comment by ftessier** [comment]
Fixed up, rebased and merge this pull request in commit b99e3b53cb7b02b7a1dba8c3579153b793e7811d
